added the first composer sample

parsagh97 opened the pull request:
some changes were made from the last demo.
the transaction between Legal and Natural person is now available.
some changes must be made in the model's design and structure.

littleali commented on lib/script.js:
```diff
@@ -0,0 +1,35 @@
+  transfer.ownership.share = transfer.share;
+  const ar = await getParticipantRegistry('org.example.empty.owner');
+  await ar.update(transfer.ownership);
+  
```
Please remove Empty lines

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,108 @@
+  o inTransfer
+  
+}
+
```
Please Remove empty lines

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,108 @@
+  o commercial
+  o land
+}
+enum transfer_status {
```
Please change it to TranferStatus

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,108 @@
+}
+
+asset Property identified by propID {
+o String propID // property ID-unique and NOT related to the No. of the property
```
Please rename to propertyId

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,108 @@
+  o String regID //registeration ID 
+  o IranAddress address
+}
+participant naturalPerson identified by natID{
```
Please rename it to NaturalPerson

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,108 @@
+  o IranAddress address
+}
+participant naturalPerson identified by natID{
+  o String natID //national ID
```
Please rename it to Id

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,108 @@
+o transfer_status stat
+}
+
+participant legalPerson identified by regID{
```
Please rename it to JuridicalPerson

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,108 @@
+}
+
+participant legalPerson identified by regID{
+  o String regID //registeration ID 
```
Please Rename it to ID

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,108 @@
+abstract concept ownership  {
+  -->Property propID 
+  o Double share
+
```
Please remove empty line

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,108 @@
+}
+
+abstract concept ownership  {
+  -->Property propID 
```
Please use propertyId

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,108 @@
+  o IranAddress address
+}
+
+abstract concept ownership  {
```
Ownership

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,108 @@
+
+}
+
+concept legOwner extends ownership{
```
JurdicalOwnership

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,108 @@
+  -->legalPerson legOwner
+}
+
+concept natOwner extends ownership{
```
NaturalOwnership

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,108 @@
+  -->naturalPerson natOwner
+}
+
+participant owner identified by OwnershipID{
```
Owner

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,108 @@
+}
+
+concept natOwner extends ownership{
+  -->naturalPerson natOwner
```
naturalOwner

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,108 @@
+}
+
+participant owner identified by OwnershipID{
+	o legOwner lOwner
```
legalOwner

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,225 @@
+  o Double Tax//in million
+}
+
+abstract concept Renting {
```
Please move this above transactions definitions

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,225 @@
+  o DateTime end
+}
+
+concept RentProperty extends Renting{
```
Please move this above transactions definitions

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,225 @@
+  o String TrackingCode
+}
+
+participant Tenant identified by TenantID {
```
You didn't use Tenant ?
If yes please remove it.

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,225 @@
+  o Double Area //in m^2
+}
+
+participant JuridicalPerson identified by regID{
```
I think we need 1 level of inheritance like this
Person-->
JurdicalPerson
NaturalPerson

Where Person is abstract participant

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,225 @@
+  o String branchID
+}
+
+transaction judStatus {
```
Please use full name

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,225 @@
+
+//--------------------------------------------------------------------------
+//judiciary
+participant judiciary identified by branchID {
```
I cant understand what does it do in our model
If we need this please change this to Judiciary

littleali commented on models/model.cto:
```diff
@@ -0,0 +1,225 @@
+  o DateTime start
+  o DateTime end
+  -->Property porp
+  -->JuridicalPerson legPerson
```
If we have Single abstract Person we have just one Person in this transaction

littleali commented on lib/script.js:
```diff
@@ -0,0 +1,151 @@
+ * @transaction
+ */
+
+async function tradeCommodity(transfer) { // eslint-disable-line no-unused-vars
```
Please remove it if it is unused

littleali commented on lib/script.js:
```diff
@@ -0,0 +1,151 @@
+    await pr.update(judiciary.natPerson);
+  } 
+  else {
+    //throw error
```
Please throw error

littleali commented on lib/script.js:
```diff
@@ -0,0 +1,151 @@
+    await ar.update(facility.prop);
+  }
+  else{
+    //throw error
```
Please throw error

littleali commented on lib/script.js:
```diff
@@ -0,0 +1,104 @@
+  async function add(AddProp){
+    const factory = getFactory();
+    const NS = 'org.example.empty';
+    const NewProperty = factory.newResource(NS , 'Property' , "1000");
```
Here you add all Properties with id = 1000 and it is imposible

littleali commented on lib/script.js:
```diff
@@ -0,0 +1,104 @@
+   async function add(test){
+    const factory = getFactory();
+    const NS = 'org.example.empty';
+    const NewProperty = factory.newResource(NS , 'test' , "100");
```
Also here you insert with static id.

parsagh97 commented:
about the transactions and the reason you mentioned that they're not working properly, I changed the static number, in this case, "1000", and replaced it with the propertyID that we had in the model file.
but still, I get the "TypeError: t[Symbol.iterator] is not a function" error, I don't know which part I'm doing wrong.

parsagh97 commented:
about some extra piece of code you mentioned in the comments, I told you about them before and why We have them there, they were added because in the MVPs paper it was mentioned and in our last talk on skype, you told me to make some changes so I just kept them
